### PR TITLE
Add MIME types for all files in packages, drop filtered file classes

### PR DESCRIPTION
**Pull request by pmatilai**
Add new tags, rpmfiles APIs and other infra to support storing and
querying file MIME types. Store MIME type for all files, stop adding
rather arbitrarily filtered file "class" data as this is bloated and
relatively useless data, remove related cruft.

Fixes: #1096

### CODE REVIEW

**Comment by pmatilai** [comment]
This is an alternative to #1099, making a clear distinction between the old class info and new MIME stuff.

File: lib/rpmfiles.h [hunk 1]
```diff
@@ -396,13 +397,21 @@ rpm_loff_t rpmfilesFSize(rpmfiles fi, int ix);
 rpm_color_t rpmfilesFColor(rpmfiles fi, int ix);
 
 /** \ingroup rpmfiles
- * Return file class from file info set.
+ * return file class from file info set.
```
Comment (Conan-Kudo) [inline lib/rpmfiles.h]: Why did you change this?

**Comment by Conan-Kudo** [comment]
I'm still somewhat uneasy about this change, but at least we're not repurposing an existing field anymore...

File: lib/rpmfiles.h (hunk 1 above)
Comment (pmatilai) [reply lib/rpmfiles.h]: By mistake, rather obviously I hope. Fixed, thanks for spotting.

**Comment by Conan-Kudo** [comment]
@pmatilai Maybe I'm blind and I missed something, but I don't seem to see an obvious source for mime data for rpm to populate MIME types for files?

**Comment by pmatilai** [comment]
Can you try explaining why this makes you uneasy? (I know well enough it's not always easy to say)
Thing is, aside from the class info being unreliable as hell, the file class data has always been optional and not all packages have it, so nothing at all can *rely* on it.

**Comment by pmatilai** [comment]
> I don't seem to see an obvious source for mime data for rpm to populate MIME types for files?
Hmm? That source was added in commit e49111d822dc1c0715f5a85ed61523b4d020ae33, this just builds on top of that.

**Comment by Conan-Kudo** [comment]
> > I don't seem to see an obvious source for mime data for rpm to populate MIME types for files?
> 
> Hmm? That source was added in commit [e49111d](https://github.com/rpm-software-management/rpm/commit/e49111d822dc1c0715f5a85ed61523b4d020ae33), this just builds on top of that. That, or I'm misunderstanding the question.

Nope, that answers the question. I forgot that was already merged earlier, which is pretty bad since I approved it! 🤦‍♂

**Comment by Conan-Kudo** [comment]
> Can you try explaining why this makes you uneasy? (I know well enough it's not always easy to say)

I think the thing that makes me uneasy is that we're simultaneously no longer populating `RPM_FILECLASS`. It's pretty rare that we do something like that. I'm semi-concerned about the quality of the classifications, but that's something that can be improved over time.

> Thing is, aside from the class info being unreliable as hell, the file class data has always been optional and not all packages have it, so nothing at all can _rely_ on it.

So, I didn't know this part. As far as I had known, file class data has always been present in RPMs built the normal way (i.e. with rpmbuild on a distro that has a functioning rpm).

**Comment by pmatilai** [comment]
Being optional might be a matter of perspective: prior to rpm 4.13, file classes and colors were not populated if external dependency generator was used, which always was a smallish but non-ignorable set of packages.

We could certainly do this in stages as well, ie first add MIME and then later axe file class, it just feels like a mountain of duplicated crud for little reason. At any rate this needs a more elaborate commit message, this is RFC status for a reason.

**Comment by Conan-Kudo** [comment]
That's a fair point, once you have a more fleshed out commit message for this, then I'd be happy look over it for final review. At least code-wise, looks good. 👍

**Comment by pmatilai** [comment]
(rebased for testing purpose, just ignore...)
